ksql-engine/src/main/java/io/confluent/ksql/util/QueuedQueryMetadata.java

```diff
@@ -67,4 +74,15 @@ public boolean equals(Object o) {
   public int hashCode() {
     return Objects.hash(rowQueue, super.hashCode());
   }
+
+  public void setLimitHandler(final OutputNode.LimitHandler limitHandler) {
+    getOutputNode().setLimitHandler(limitHandler);
+  }
+
+  private class StateListener implements KafkaStreams.StateListener {
```

Comment:
Not for this PR, but this may be worth implementing for the `PersistentQueryMetadata` as well, and then update the metrics for the number of persistent queries, if a query stops running all of a sudden.

Reply:
Agreed.